jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/ThreadPool.java

```diff
@@ -112,21 +119,19 @@ static void executeImmediately(Executor executor, Runnable task)
             return;
         }
 
-        switch (Invocable.getInvocationType(task))
+        if (invocationType == Invocable.InvocationType.EITHER)
+        {
+            Invocable.invokeNonBlocking(task);
+            return;
+        }
```

Comment:
This too could be moved to be before the tryExecute or virtual.    But should seldom/never happen

Reply:
Because `EITHER` may be blocking, we want to try to execute it first... 
Actually, I think we can remove this code, as we are going to run it for sure (a requisite of `EITHER`) in another thread, it is never queued.

Reply:
But if we invokeNonBlocking, then it will not block and acts just like a NON_BLOCKING task.